core/controlplane/config/templates/cloud-config-controller

```diff
@@ -3569,6 +3592,253 @@ write_files:
         namespace: kube-system
 {{end}}
 
+{{if .Experimental.KIAMSupport.Enabled }}
+  - path: /etc/kubernetes/ssl/kiam/ca.pem
+    encoding: gzip+base64
+    content: {{.AssetsConfig.KIAMCACert}}
+
+  - path: /etc/kubernetes/ssl/kiam/server-key.pem{{if .AssetsEncryptionEnabled}}.enc{{end}}
+    encoding: gzip+base64
+    content: {{.AssetsConfig.KIAMServerKey}}
+
+  - path: /etc/kubernetes/ssl/kiam/server.pem
+    encoding: gzip+base64
+    content: {{.AssetsConfig.KIAMServerCert}}
+
+  - path: /etc/kubernetes/ssl/kiam/agent-key.pem{{if .AssetsEncryptionEnabled}}.enc{{end}}
+    encoding: gzip+base64
+    content: {{.AssetsConfig.KIAMAgentKey}}
+
+  - path: /etc/kubernetes/ssl/kiam/agent.pem
+    encoding: gzip+base64
+    content: {{.AssetsConfig.KIAMAgentCert}}
+
+  # Without the namespace annotation the pod will be unable to assume any roles
+  - path: /srv/kubernetes/manifests/kube-system-ns.yaml
+    content: |
+      apiVersion: v1
+      kind: Namespace
+      metadata:
+        name: kube-system
+        annotations:
+          iam.amazonaws.com/permitted: ".*"
+  - path: /srv/kubernetes/manifests/kiam-all.yaml
+    content: |
+      apiVersion: extensions/v1beta1
+      kind: DaemonSet
+      metadata:
+        namespace: kube-system
+        name: kiam-server
+      spec:
+        template:
+          metadata:
+            annotations:
+              prometheus.io/scrape: "true"
+              prometheus.io/port: "9620"
+            labels:
+              app: kiam
+              role: server
+          spec:
+            {{if .Experimental.Admission.Priority.Enabled -}}
+            priorityClassName: system-node-critical
+            {{ end -}}
+            tolerations:
+            - operator: Exists
+              effect: NoSchedule
+            - operator: Exists
+              effect: NoExecute
+            - operator: Exists
+              key: CriticalAddonsOnly
+            serviceAccountName: kiam-server
+            nodeSelector:
+              kubernetes.io/role: master
```

Comment:
Ha, just realized that kiam's default template spec relies on `kubernetes.io/role: master`.
kube-aws has been using `node-role.kubernetes.io/master: ""`.
Both seems valid according to: https://github.com/kubernetes/kubernetes/issues/50010 and https://github.com/kubernetes/kubernetes/pull/51173